lib/ex_gram/dispatcher.ex

```diff
@@ -51,63 +51,92 @@ defmodule ExGram.Dispatcher do
     {:ok, ops}
   end
 
-  def handle_call(
-        {:update, u},
-        _from,
-        %{
-          handler: handler,
-          error_handler: error_handler
-        } = s
-      ) do
-    cnt = create_cnt(s) |> Map.put(:update, u)
-
-    case apply_middlewares(cnt) do
-      %Cnt{halted: true} ->
-        # Logger.info "Middleware cancel"
-        true
-
-      cnt ->
-        info = extract_info(cnt)
-        spawn(fn -> call_handler(handler, info, cnt, error_handler) end)
-    end
+  def handle_call({:update, u}, _from, %{handler: handler, error_handler: error_handler} = state) do
+    cnt =
+      state
+      |> create_cnt()
+      |> Map.put(:update, u)
+
+    meta = %{context: cnt}
 
-    {:reply, :ok, s}
+    with_span(meta, fn ->
+      case apply_middlewares(cnt) do
+        %Cnt{halted: true} ->
+          {nil, %{result: :halted}}
+
+        cnt ->
+          info = extract_info(cnt)
+          spawn(fn -> call_handler(handler, info, cnt, error_handler) end)
```

Comment:
All the handlers executed with spawn are not being reported correctly 🤔  
The easy solution would be to add another span on the call_handler